### ISSUE TITLE
Handle routing / razor pages configuration internally

### ISSUE THREAD

**Issue by brianweet** [Geta/geta-optimizely-sitemaps#38]
We now require projects to configure razor pages. This should be something we handle inside the package.
Next to the fact that it is slightly annoying, an actual issue occurs when a project has custom configuration for razor pages.
The package should still work regardless of the project configuration.

Look into official packages to see how they do it; they have a controller and some additional module set up. (e.g. EPiServer.Cms.UI.Admin)

**Comment by marisks** [comment]
I disagree, Razor pages are framework dependency. We should not do it ourselves. This is a pre-requisite like any other pre-requisites.

**Comment by jevgenijsp** [comment]
I think we are faced with something similar. 
For example, if our project is configured with a custom razor pages folder root directory:
```
services.AddRazorPages()
    .WithRazorPagesRoot("/Features/");
```

Sitemaps won't work because it overrides the rule that razor pages are stored in '/Pages' folder.

**Comment by marisks** [comment]
@jevgenijsp I think Brian means - calling AddRazorPages internally if it is not already called.

**Comment by jevgenijsp** [comment]
Yeah. And maybe this resolves the case with razor pages root directory overriding from an external project.

**Comment by marisks** [comment]
@jevgenijsp That's a different issue. I will check how to solve it. Probably the best would be to generate links based on razor config. But it is not possible to have different root directories for different libraries - the root directory is common for all Razor pages.
@brianweet I looked at `EPiServer.Cms.UI.Admin` and they do not have Razor pages. Also, they explicitly require calling `MapContent` under `UseEndpoints`. So it is the same as requiring calling `MapRazorPages`. And in the latest examples (Foundation), they included `MapRazorPages` call too. So I think it will not be an issue in the future.

**Comment by brianweet** [comment]
@marisks I think the package should work regardless of project configuration. Exactly what @jevgenijsp is referring to; configuring razor pages the way you like, in your own project, should be possible.
Therefore I'd opt for adding explicit configuration for mapping/routing for our razor pages inside the package somehow. I thought maybe `EPiServer.Cms.UI.Admin` would provide an example but too bad that it's not.
I will ask Opti directly, perhaps they can give us some pointers.

**Comment by marisks** [comment]
@brianweet Looked at @jevgenijsp issue and found that it is not possible to configure Razor Pages root directory in a scoped manner - so that our library could use our own root directory. It is a singleton configuration where they use a "root directory" for all Razor pages.

I think enabling Razor Pages is a prerequisite for using our library. And as more and more are using Razor pages as default, that prerequisite will be automatically done. 

Another issue is what @jevgenijsp found, but the only thing we could do is to create an example and issue a bug report on ASP.NET Core repo.

**Comment by brianweet** [comment]
@marisks I agree on having Razor Pages as prerequisite for the library. Too bad there's no way to override the root directory or have multiple root directories somehow. Sounds like a limitation of ASP.NET indeed.
Then, at least for now, another prerequisite is not to customize the Razor Pages root directory in your own project.

**Comment by brianweet** [comment]
WithRazorPagesRoot issue is fixed by switching to Areas
And I agree that razor pages is a pre-requisite to using the lib
